build/v1/types.go

```diff
@@ -1022,23 +1026,37 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// BuildImageChangeTriggerKey captures the required elements for identifying the ImageStreamTag referenced by the more
+// generic ObjectReference BuildTriggerPolicy.ImageChange.From.  It is used by ImageChangeTriggerStatus, where a
+// specific instance of ImageChangeTriggerStatus in maintained in BuildConfigStatus.ImageChangeTriggers for each entry
+// in the BuildConfigSpec.Triggers array where the BuildTriggerPolicy.ImageChange pointer is set to a non-nil value
+type BuildImageChangeTriggerKey struct {
+	// namespace is the namespace where the ImageStreamTag used for an ImageChangeTrigger is located
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
+
+	// name is the name of the ImageStreamTag used for an ImageChangeTrigger
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+}
+
 // ImageChangeTriggerStatus tracks the latest resolved status of the associated ImageChangeTrigger policy
 // specified in the BuildConfigSpec.Triggers struct.
 type ImageChangeTriggerStatus struct {
-	// lastTriggeredImageID is the sha/id of the imageref cited in the 'from' field the last time this BuildConfig was triggered.
+	// lastTriggeredImageID represents the current sha/id of the ImageStreamTag cited in the 'from' field of a corresponding
```

Comment:
current from what perspective?  Last one used when the new build was created somehow?

Reply:
yes, the value at the time the buildconfig most recently instantiated a build.

@gabemontero the current text isn't quite correct if i recall the behavior:

> the last time a Build for this BuildConfig was instantiated as a result of *ANY* of the ImageChangeTriggers
in the BuildConfigSpec.Triggers array firing

it's not just when a build was instantiated as a result of a trigger firing.  A webhook, manual invocation, commit hook, etc all have the same behavior.  Any time the buildconfig runs a build, these imageids all get updated to their current values (reflecting the SHA value they had at the time the last build was run, and ensuring we don't immediately run a 2nd build for the same SHA)

Reply:
OK I will add the "instantiate in any way not just ICT" element @bparees 

I was trying to convey the `current from what perspective` element @deads2k asked about ... I'll take another stab at making that clearer.

Both in the next push.